pygfx/objects/_base.py

```diff
@@ -7,9 +7,12 @@
 from ..resources import Resource, Buffer
 from ..utils import array_from_shadertype
 
-# Keep track of id's
+# Keep track of id's. About the max:
+# * 2_147_483_647 (2**31 -1) max number for signed i32.
+# *    16_777_216 max integer that can be stored exactly in f32
+# *     4_000_000 max integer that can survive being passed as a varying
```

Comment:
Why?

Reply:
I tested it. I expected simple float rounding, but apparently magic stuff happens there ... I just hope this is not too device-specific ... Worst case we can use 2 values to encode one int. But then we need two varyings, because we do not have `vec5` :)